include/zephyr/bluetooth/ead.h

```diff
@@ -0,0 +1,147 @@
+ * - MIC is added in the last 4 bytes.
+ *
+ * @attention The function must be called each time the RPA is updated or the
+ * data are modified.
+ *
+ * @note The term `advertising structure` is used to describe the advertising
+ *       data with the advertising type and the length of those two.
+ *
+ * @param[in]  session_key Key of 16 bytes used for the encryption.
+ * @param[in]  iv Initialisation Vector used to generate the nonce. It must be
+ *             changed each time the Session Key change.
+ * @param[in]  payload Advertising Data to encrypt. Can be multiple advertising
+ *             structure concatenated.
+ * @param[in]  payload_size Size of the Advertising Data to encrypt.
+ * @param[out] encrypted_payload Encrypted Adv Data including the Randomizer and
+ *             the MIC. Size must be at least `BT_EAD_RANDOMIZER` + @p
+ *             payload_size + `BT_EAD_MIC_SIZE`.
+ *
+ * @retval 0 Data have been correctly encrypted and authenticated.
+ * @retval -EIO Error occurred during the encryption or the authentication.
+ * @retval -EINVAL One of the argument is a NULL pointer.
+ * @retval -ECANCELED Error occurred during the random number generation.
+ */
+int bt_ead_encrypt(const uint8_t *session_key, const uint8_t *iv, const uint8_t *payload,
+		   size_t payload_size, uint8_t *encrypted_payload);
+
+/**
+ * @brief Decrypt and authenticate the given encrypted advertising data.
+ *
+ * @note The term `advertising structure` is used to describe the advertising
+ *       data with the advertising type and the length of those two.
+ *
+ * @param[in]  session_key Key of 16 bytes used for the encryption.
+ * @param[in]  iv Initialisation Vector used to generate the `nonce`.
+ * @param[in]  encrypted_payload Encrypted Advertising Data received. This
+ *             should only contain the advertising data from the received
+ *             advertising structure, not the length nor the type.
+ * @param[in]  encrypted_payload_size Size of the received advertising data in
+ *             bytes. Should be equal to the length field of the received
+ *             advertising structure, minus the size of the type (1 byte).
+ * @param[out] payload Decrypted advertising payload.
+ *
+ * @retval 0 Data have been correctly decrypted and authenticated.
+ * @retval -EIO Error occurred during the decryption or the authentication.
+ * @retval -EINVAL One of the argument is a NULL pointer or @p encrypted_payload_size is less than
+ *                 BT_EAD_RANDOMIZER_SIZE + BT_EAD_MIC_SIZE.
+ */
+int bt_ead_decrypt(const uint8_t *session_key, const uint8_t *iv, const uint8_t *encrypted_payload,
+		   size_t encrypted_payload_size, uint8_t *payload);
+
+/**
+ * @brief Helper function to get the size (in bytes) of the encrypted
+ * advertising data for a given set of `bt_data` structure.
+ *
+ * @param[in] ad_structs Array of `bt_data` structure.
+ * @param[in] num_elts Number of `bt_data` structure in @p ad_structs.
```

Comment:
What is `elts` short for?

Reply:
Elements, I can write it completely, it will be more readable.

Reply:
What about just `count`, `cnt, `length` or `len`, which are typically used for this purpose in the existing BT API?